src/bonefish/messages/wamp_call_message.hpp

```diff
@@ -185,7 +185,7 @@ inline void wamp_call_message::set_options(const msgpack::object& options)
 
 inline void wamp_call_message::set_procedure(const std::string& procedure)
 {
-    m_procedure = procedure;
+    m_procedure = msgpack::object(procedure, get_zone());
```

Comment:
It is part of the call message class interface for completeness. If we had unit tests then there would be unit test coverage for this. It is a string so it always needs to have a zone provided in order for msgpack to work correctly. Clearly there is a bug in msgpack-c since the following statement compiles and is obviously wrong:

```
#include <msgpack.hpp>
#include <string>

msgpack::object obj;

void set_procedure(const std::string procedure)
{
    obj = procedure;
}

int main(int argc, char** argv)
{
    set_procedure("foo");
    return 0;
}
```